src/parser.rs

```diff
@@ -76,26 +75,37 @@ impl<'a, R: Read> Parser<'a, R> {
     /// Expects that the current token is an opening tag like `<tag>` without attributes (and a
     /// local name without namespaces). If not, returns an error.
     fn expect_open(&mut self, tag: &str) -> ParseResult<()> {
-        match self.cur.clone() {
-            XmlEvent::StartElement { ref name, .. }
-            if name == &OwnedName::local(tag) => {
-                self.next()?;
-                Ok(())
-            }
+        match self.cur {
+            XmlEvent::StartElement { ref name, .. } if name.local_name == tag => {},
```

Comment:
I originally did this the other way because it takes prefix and namespace into account, this will also accept things like `<my:value>`, which I'd like to reject. Unfortunately `xml-rs` doesn't offer convenient `ParitalEq` implementations, but you can call `borrow()` on the `OwnedName` to get a borrowed `Name`, and call `Name::local(tag)` to get a `Name` without having to allocate.

Reply:
Yeah. This is a controversial change but `next()` cares about namespaces now.

Reply:
AFAIK they're not allowed in XML-RPC, or have you seen something using them? If not, I think I'd like to continue to reject them. You can also do that inside `next`, just match on `StartElement` and return an error when `namespace` or `prefix` of the `OwnedName` are set.

Reply:
I mean, that now `next()` rejects non-local tags. That's why comparing `name.local_name` only is safe.

Reply:
Oooh I see. Not sure why added the check to `next` but kept the `OwnedName` comparison... In that case, looks good!